gate-plugins/src/main/kotlin/com/netflix/spinnaker/gate/plugins/web/installed/PluginsInstalledController.kt

```diff
@@ -0,0 +1,86 @@
+import io.swagger.annotations.ApiOperation
+import java.util.stream.Collectors
+import org.pf4j.PluginWrapper
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestMethod
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+import retrofit.RetrofitError
+
+@RestController
+@RequestMapping("/plugins/installed")
+class PluginsInstalledController(
+  private val clouddriverService: ClouddriverService,
+  private val echoService: EchoService,
+  private val fiatService: ExtendedFiatService,
+  private val front50Service: Front50Service,
+  private val igorService: IgorService,
+  private val keelService: KeelService,
+  private val orcaServiceSelector: OrcaServiceSelector,
+  private val roscoService: RoscoService,
+  private val swabbieService: SwabbieService,
+  private val spinnakerPluginManager: SpinnakerPluginManager
+) {
+
+  @ApiOperation(value = "Get all installed Spinnaker plugins")
+  @RequestMapping(method = [RequestMethod.GET])
+  fun getInstalledPlugins(@RequestParam(value = "service", required = false) service: String?): Map<String, List<SpinnakerPluginDescriptor>> {
+    return when (service) {
+      "clouddriver" -> mutableMapOf(Pair(service, callService { clouddriverService.installedPlugins }))
+      "echo" -> mutableMapOf(Pair(service, callService { echoService.installedPlugins }))
+      "fiat" -> mutableMapOf(Pair(service, callService { fiatService.installedPlugins }))
+      "front50" -> mutableMapOf(Pair(service, callService { front50Service.installedPlugins }))
+      "gate" -> mutableMapOf(Pair(service, gatePlugins()))
+      "igor" -> mutableMapOf(Pair(service, callService { igorService.installedPlugins }))
+      "keel" -> mutableMapOf(Pair(service, callService { keelService.installedPlugins }))
+      "orca" -> mutableMapOf(Pair(service, callService { orcaServiceSelector.select().installedPlugins }))
+      "rosco" -> mutableMapOf(Pair(service, callService { roscoService.installedPlugins }))
+      "swabbie" -> mutableMapOf(Pair(service, callService { swabbieService.installedPlugins }))
+      else -> {
+        mutableMapOf(
+          Pair("clouddriver", callService { clouddriverService.installedPlugins }),
+          Pair("echo", callService { echoService.installedPlugins }),
+          Pair("fiat", callService { fiatService.installedPlugins }),
+          Pair("front50", callService { front50Service.installedPlugins }),
+          Pair("gate", gatePlugins()),
+          Pair("igor", callService { igorService.installedPlugins }),
+          Pair("keel", callService { keelService.installedPlugins }),
+          Pair("orca", callService { orcaServiceSelector.select().installedPlugins }),
+          Pair("rosco", callService { roscoService.installedPlugins }),
+          Pair("swabbie", callService { swabbieService.installedPlugins })
+        ).toSortedMap()
+      }
+    }
+  }
```

Comment:
Retrofit interfaces can not be extended, so all the service calls have to be explicit per service interface.

Reply:
I ran into that as well, super annoying!